File: test-suite/handwritten-src/js/WcharTest.js
class WcharTest {
    constructor(module) {
        this.m = module;
    }

    static STR1 = "some string with unicode \u{0000}, \u{263A}, \u{D83D}\u{DCA9} symbols";
    static STR2 = "another string with unicode \u{263B}, \u{D83D}\u{DCA8} symbols";

    test() {
        assertEq(this.m.testsuite.WcharTestHelpers.getRecord().s, WcharTest.STR1);
        assertEq(this.m.testsuite.WcharTestHelpers.getString(), WcharTest.STR2);
        assertEq(this.m.testsuite.WcharTestHelpers.checkString(WcharTest.STR2), true);
        assertEq(this.m.testsuite.WcharTestHelpers.checkRecord({s: WcharTest.STR1}), true);
    }
}

allTests.push(WcharTest);

File: test-suite/handwritten-src/swift/WcharTest.swift
import XCTest
@testable import TestSuite

final class WcharTest: XCTestCase {
    let STR1: String = "some string with unicode \u{0000}, \u{263A}, 💩 symbols";
    let STR2: String = "another string with unicode \u{263B}, 💨 symbols";

    func test() throws {
        XCTAssertEqual(try WcharTestHelpers_statics.getRecord().s, STR1)
        XCTAssertEqual(try WcharTestHelpers_statics.getString(), STR2)
        XCTAssertTrue(try WcharTestHelpers_statics.checkString(STR2))
        XCTAssertTrue(try WcharTestHelpers_statics.checkRecord(WcharTestRec(s:STR1)))
    }
}
